rootfs/rootfs/usr/local/etc/init.d/docker

```diff
@@ -114,10 +114,12 @@ stop() {
 
 restart() {
     if check; then
-        stop && sleep 1 && start
-    else
-        start
+        stop
+        while check ; do
+            sleep 1
+        done
```

Comment:
If the Docker daemon somehow doesn't get signaled properly during `stop`, this will essentially be an infinite loop -- do you think it warrants adding some simple "max tries" logic to give us a timeout?